File: backend/app/schemas/subscription.py
from datetime import datetime
from typing import Optional
from app.schemas.common import CamelModel


class UpgradeRequest(CamelModel):
    """구독 업그레이드 요청 — PRO/ENTERPRISE는 payment_key 필수"""
    plan: str
    payment_key: str | None = None


class SubscriptionResponse(CamelModel):
    """현재 구독 정보 응답"""
    id: str
    plan: str
    status: str
    started_at: datetime
    expires_at: Optional[datetime] = None


class PlanInfo(CamelModel):
    """요금제 정보"""
    name: str
    display_name: str
    price: int
    currency: str
    billing_period: str
    features: list[str]
    is_popular: bool


class PlansListResponse(CamelModel):
    """요금제 목록 응답"""
    plans: list[PlanInfo]

File: vxmi-dashboard/src/types/subscription.ts
/** 구독/요금제 관련 타입 */

// 구독 상태
export type SubscriptionStatus = 'ACTIVE' | 'EXPIRED' | 'CANCELLED' | 'TRIAL';

// 현재 구독 정보
export interface SubscriptionResponse {
  id: string;
  plan: string;
  status: SubscriptionStatus;
  startedAt: string;
  expiresAt?: string;
}

// 요금제 정보
export interface PlanInfo {
  name: string;
  displayName: string;
  price: number;
  currency: string;
  billingPeriod: string;
  features: string[];
  isPopular: boolean;
}

// 요금제 목록 응답
export interface PlansListResponse {
  plans: PlanInfo[];
}
